### PR TITLE
Issue

### PR DESCRIPTION
**Pull request by PaulinaKoltun**

### CODE REVIEW

File: src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java [hunk 1]
```diff
@@ -81,12 +81,21 @@ public void saveRoom(Room room) {
     }
 
     @Override
-    public List<Room> getAllRooms(Pageable pageable) {
+    public List<Room> getAllRooms(String type, Pageable pageable) {
         Page<RoomEntity> roomEntityList = roomRepository.findAll(pageable);
```
Comment (robertszczygielski) [inline src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java]: You can put  'if' here and in roomRepository create two method e.g.: one - findAllPlantRoom, second - findAllPetRoom, the sql query should grab all rooms.

File: src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java [hunk 2]
```diff
@@ -81,12 +81,21 @@ public void saveRoom(Room room) {
...
 
-        return roomEntityList.getContent().stream()
-                .map(r -> roomMap.map(r))
-                .collect(Collectors.toList());
+        if ("plant".equals(type)) {
```
Comment (robertszczygielski) [inline src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java]: Redundant 'if' collection should by returned by database.

File: src/test/java/org/pethotel/HeavenForPets/service/impl/AddressServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    private AddressRepository addressRepository;
+
+    @Mock
+    private AddressMap addressMap;
+
+    @InjectMocks
+    private AddressServiceImpl addressService;
+
+    @Test
+    public void saveAddress() throws Exception {
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/AddressServiceImplTest.java]: name of method, what the method tests, e.g. checkIfRepossorySaveMethodWasCalled

File: src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    private FoodRepository foodRepository;
+
+    @Mock
+    private MainGeneratorChain mainGeneratorChain;
+
+    @InjectMocks
+    private FoodServiceImpl foodService;
+
+    @Test
+    public void saveFoodWhenNull() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java]: name, e.g. checkIfRepositorySaveMethodWasCalled

File: src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        when(foodMap.map(food)).thenReturn(newFoodEntity);
+        when(foodRepository.save(newFoodEntity)).thenReturn(newFoodEntity);
+
+        foodService.saveFood(foodlist);
+
+        verify(foodRepository, times(1)).save(newFoodEntity);
+    }
+
+    @Test
+    public void saveFoodWhenNotNull() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java]: name, e.g. checkIfRepositorySaveMethodWasCalledWhenFoonNotNull

File: src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                food.getFoodType(), food.getPetType(), food.getTaste())).thenReturn(foodEntity);
+        when(foodRepository.save(foodEntity)).thenReturn(foodEntity);
+
+        foodService.saveFood(foodlist);
+
+        verify(foodRepository, times(1)).save(foodEntity);
+    }
+
+    @Test
+    public void getFoodByPet() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java]: name, e.g. checkIfMapFromEntityToDtoWassCalled

File: src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        when(foodRepository.getFoodByPetType(petType)).thenReturn(foodEntityList);
+        when(foodMap.map(foodEntity)).thenReturn(food);
+
+        foodService.getFoodByPet("FISH");
+
+        verify(foodMap, times(1)).map(foodEntity);
+    }
+
+    @Test
+    public void getFoodDetailsById() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java]: This method is really needed? You should check if repository was called and name e.g. checkIfRepositoryFindOneWasCalled

File: src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        FoodEntity foodEntity = new FoodEntity();
+
+        when(foodRepository.findOne((long) id)).thenReturn(foodEntity);
+
+        FoodDetails foodDetails = foodService.getFoodDetailsById(id);
+
+        assertEquals(foodEntity.getDeliveryAmount(), foodDetails.getDeliveryAmount());
+    }
+
+    @Ignore
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java:107]: it blows my mind

File: src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+        HttpServletResponse response = null;
+
+        FoodDetails foodDetails = foodService.getFoodDetailsById(id);
+
+        mainGeneratorChain.makeResponse(request, response, file, foodDetails);
+
+    }
+
+    @Test
+    public void getFoodById(){
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/FoodServiceImplTest.java]: name, e.g checkIdRepositoryFindOneMethod WasCalled

File: src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+    private PetMap petMap;
+
+    @Mock
+    private AddressService addressService;
+
+    @InjectMocks
+    private OwnerServiceImpl ownerService;
+
+    @Test
+    public void saveOwner() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+        when(ownerMap.map(owner)).thenReturn(ownerEntity);
+        when(ownerRepository.save(ownerEntity)).thenReturn(ownerEntity);
+
+        ownerService.saveOwner(owner);
+
+        verify(ownerRepository, times(1)).save(ownerEntity);
+    }
+
+    @Test
+    public void getAllClients(){
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+        when(ownerRepository.findAll()).thenReturn(ownerEntityList);
+        when(clientMap.map(ownerEntity)).thenReturn(client);
+
+        ownerService.getAllClients();
+
+        verify(clientMap, times(1)).map(ownerEntity);
+    }
+
+    @Test
+    public void showAllPets(){
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+        when(ownerRepository.findOne((long)id)).thenReturn(ownerEntity);
+        when(petMap.map(petEntity)).thenReturn(pet);
+
+        ownerService.showAllPets(id);
+
+        verify(petMap, times(1)).map(petEntity);
+    }
+
+    @Test
+    public void pickupAllPets(){
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+
+        when(ownerRepository.findOne(Long.valueOf(id))).thenReturn(ownerEntity);
+
+        ownerService.pickupAllPets(id);
+
+        assertEquals(null, petEntity.getRoomEntity());
+    }
+
+    @Test
+    public void updateDiscountAtOwner(){
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+        when(ownerRepository.getOwnerByLastName(lastName)).thenReturn(ownerEntity);
+        when(ownerRepository.save(ownerEntity)).thenReturn(ownerEntity);
+
+        ownerService.updateDiscountAtOwner(lastName, discount);
+
+        verify(ownerRepository, times(1)).save(ownerEntity);
+    }
+
+    @Test
+    public void getOwnerById(){
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/OwnerServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+
+    @Mock
+    private PetMap petMap;
+
+    @InjectMocks
+    private PetServiceImpl petService;
+
+
+    @Test
+    public void getPets() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+
+        when(petRepository.findAll()).thenReturn(petEntities);
+
+        petService.getPets();
+
+        verify(petRepository, times(1)).findAll();
+    }
+
+    @Test
+    public void bringPetAgain() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+        when(petMap.map(pet, petEntity, roomEntity)).thenReturn(petEntity);
+        when(petRepository.save(petEntity)).thenReturn(petEntity);
+
+        petService.bringPetAgain(pet);
+
+        verify(petRepository, times(1)).save(petEntity);
+    }
+
+    @Test
+    public void pickUpPets() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+        when(petRepository.findOne((long) id)).thenReturn(petEntity);
+        when(petRepository.save(petEntity)).thenReturn(petEntity);
+
+        petService.pickupPets(idList);
+
+        verify(petRepository,times(1)).save(petEntity);
+    }
+
+    @Test
+    public void addPettoOwner() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/PetServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 20]
```diff
@@ -0,0 +1,212 @@
...
+    private RoomMap roomMap;
+
+    @Mock
+    private ShelfMap shelfMap;
+
+    @InjectMocks
+    private RoomServiceImpl roomService;
+
+    @Test
+    public void saveRoomWhenPetRoom() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 21]
```diff
@@ -0,0 +1,212 @@
...
+        when(roomRepository.getRoomByNumber(room.getRoomNumber())).thenReturn(null);
+        when(roomRepository.save(roomEntity)).thenReturn(roomEntity);
+
+        roomService.saveRoom(room);
+
+        verify(roomRepository, times(1)).save(roomEntity);
+    }
+
+    @Test
+    public void saveRoomWhenPlantRoom() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 22]
```diff
@@ -0,0 +1,212 @@
...
+        when(shelfRepository.save(shelfEntity)).thenReturn(shelfEntity);
+        when(roomRepository.save(roomEntity)).thenReturn(roomEntity);
+
+        roomService.saveRoom(plantRoom);
+
+        verify(shelfRepository, times(1)).save(shelfEntity);
+    }
+
+    @Test
+    public void saveRoomWhenNotExist() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 23]
```diff
@@ -0,0 +1,212 @@
...
+        PetRoom room = new PetRoom();
+        RoomEntity roomEntity = new RoomEntity();
+
+        when(roomRepository.getRoomByNumber(room.getRoomNumber())).thenReturn(roomEntity);
+
+        roomService.saveRoom(room);
+    }
+
+    @Test
+    public void getAllNumbers() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 24]
```diff
@@ -0,0 +1,212 @@
...
+
+        when(roomRepository.findAll()).thenReturn(roomEntityList);
+
+        roomService.getAllNumbers();
+
+        verify(roomRepository, times(1)).findAll();
+    }
+
+    @Test
+    public void getAllRooms() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 25]
```diff
@@ -0,0 +1,212 @@
...
+        when(roomRepository.findAll(pageable)).thenReturn(roomEntityList);
+        when(roomMap.map(roomEntity)).thenReturn(room);
+
+        roomService.getAllRooms(type, pageable);
+
+        verify(roomMap, times(1)).map(roomEntity);
+    }
+
+    @Test
+    public void findByRoomNumber() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 26]
```diff
@@ -0,0 +1,212 @@
...
+
+        when(roomRepository.getRoomByNumber(roomNumber)).thenReturn(roomEntity);
+
+        roomService.findByRoomNumber(roomNumber);
+
+        verify(roomRepository, times(1)).getRoomByNumber(roomNumber);
+    }
+
+    @Test
+    public void deleteRoom() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 27]
```diff
@@ -0,0 +1,212 @@
...
+
+        when(roomRepository.getRoomByNumber(roomNumber)).thenReturn(roomEntity).thenReturn(null);
+
+        roomService.deleteRoom(roomNumber);
+
+        verify(roomRepository, times(1)).delete(roomEntity);
+    }
+
+    @Test
+    public void updateRoom() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 28]
```diff
@@ -0,0 +1,212 @@
...
+        when(roomMap.map(roomEntity, room)).thenReturn(roomEntity);
+        when(roomRepository.save(roomEntity)).thenReturn(roomEntity);
+
+        roomService.updateRoom(room);
+
+        verify(roomMap, times(1)).map(roomEntity, room);
+    }
+
+    @Test
+    public void getAllRoomsByPetType() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 29]
```diff
@@ -0,0 +1,212 @@
...
+        when(roomRepository.findAll()).thenReturn(roomEntityList);
+        when(roomMap.map(roomEntity)).thenReturn(room);
+
+        roomService.getAllRoomsByType(petType);
+
+        verify(roomMap, times(1)).map(roomEntity);
+    }
+
+    @Test
+    public void freePlacesForPetType() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name, what verify this method?

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 30]
```diff
@@ -0,0 +1,212 @@
...
+        roomEntity.setFreePlaces(2);
+        roomEntityList = Arrays.asList(roomEntity, roomEntity);
+
+        when(roomRepository.findAll()).thenReturn(roomEntityList);
+
+        roomService.freePlacesForPetType();
+    }
+
+    @Test
+    public void getNumberOfRooms() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 31]
```diff
@@ -0,0 +1,212 @@
...
+
+        when(roomRepository.findAll()).thenReturn(roomEntityList);
+
+        roomService.getNumberOfRooms();
+
+        verify(roomRepository, times(1)).findAll();
+    }
+
+    @Test
+    public void getRoomByNumber() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImplTest.java [hunk 32]
```diff
@@ -0,0 +1,92 @@
...
+
+    @Mock
+    private RoomService roomService;
+
+    @InjectMocks
+    private ShelfServiceImpl shelfService;
+
+
+    @Test
+    public void saveShelves() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImplTest.java [hunk 33]
```diff
@@ -0,0 +1,92 @@
...
+        when(roomService.getRoomByNumber(id)).thenReturn(roomEntity);
+        when(shelfMap.map(shelf)).thenReturn(entity);
+
+        shelfService.saveShelves(shelves, id);
+
+        verify(shelfMap, times(1)).map(shelf);
+    }
+
+    @Test
+    public void getShelvesFromRoom() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImplTest.java]: name

File: src/test/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImplTest.java [hunk 34]
```diff
@@ -0,0 +1,92 @@
...
+        when(roomService.getRoomByNumber(id)).thenReturn(roomEntity);
+        when(shelfMap.map(shelfEntity)).thenReturn(shelf);
+
+        shelfService.getShelvesFromRoom(id);
+
+        verify(shelfMap, times(1)).map(shelfEntity);
+    }
+
+    @Test
+    public void getFreeShelves() throws Exception{
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImplTest.java]: name

File: src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java (hunk 1 above)
Comment (PaulinaKoltun) [reply src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java]: done

**Pull request merged**